test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,204 @@
+        // threads per block while this kernel is being run.
+
+        // TODO: Use function templates to parametrize and shorten the code!
+        double var3 = var2 + fmod(var2, 5);
+        double var4 = var3 + fmod(var3, 5);
+        double var5 = var4 + fmod(var4, 5);
+        double var6 = var5 + fmod(var5, 5);
+        double var7 = var6 + fmod(var6, 5);
+        double var8 = var7 + fmod(var7, 5);
+        double var9 = var8 + fmod(var8, 5);
+        double var10 = var9 + fmod(var9, 5);
+        double var11 = var10 + fmod(var10, 5);
+        double var12 = var11 + fmod(var11, 5);
+        double var13 = var12 + fmod(var12, 5);
+        double var14 = var13 + fmod(var13, 5);
+        double var15 = var14 + fmod(var14, 5);
+        double var16 = var15 + fmod(var15, 5);
+        double var17 = var16 + fmod(var16, 5);
+        double var18 = var17 + fmod(var17, 5);
+        double var19 = var18 + fmod(var18, 5);
+        double var20 = var19 + fmod(var19, 5);
+        double var21 = var20 + fmod(var20, 5);
+        double var22 = var21 + fmod(var21, 5);
+        double var23 = var22 + fmod(var22, 5);
+        double var24 = var23 + fmod(var23, 5);
+        double var25 = var24 + fmod(var24, 5);
+        double var26 = var25 + fmod(var25, 5);
+        double var27 = var26 + fmod(var26, 5);
+        double var28 = var27 + fmod(var27, 5);
+        double var29 = var28 + fmod(var28, 5);
+        double var30 = var29 + fmod(var29, 5);
+        double var31 = var30 + fmod(var30, 5);
+        double var32 = var31 + fmod(var31, 5);
+        double var33 = var32 + fmod(var32, 5);
+        double var34 = var33 + fmod(var33, 5);
+        double var35 = var34 + fmod(var34, 5);
+
+        double sum = var0 + var1 + var2 + var3 + var4 + var5 + var6 + var7 + var8 + var9 + var10 + var11 + var12
+                     + var13 + var14 + var15 + var16 + var17 + var18 + var19 + var20 + var21 + var22 + var23 + var24
+                     + var25 + var26 + var27 + var28 + var29 + var30 + var31 + var32 + var33 + var34 + var35;
+        printf("The sum is %5.2f, the argument is %lu ", sum, sizeOfArray);
+    }
+};
+
+using TestAccs = alpaka::test::EnabledAccs<alpaka::DimInt<1u>, std::uint32_t>;
+
+TEMPLATE_LIST_TEST_CASE("getFunctionAttributes.1D.withIdx", "[workDivKernel]", TestAccs)
+{
+    using Acc = TestType;
+    using Idx = alpaka::Idx<Acc>;
+    using Dim = alpaka::Dim<Acc>;
+    using Vec = alpaka::Vec<Dim, Idx>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+    auto const platform = alpaka::Platform<Acc>{};
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    auto const& bundeledKernel = alpaka::KernelBundle(kernel, 200ul);
+
+    // Test getValidWorkDivForKernel for 1024 threads per grid
+    auto const workDiv = alpaka::getValidWorkDivForKernel<Acc>(dev, bundeledKernel, Vec{1024}, Vec{1});
```

Comment:
IMO you need to query the maximum number of threads the divice can handle instead of hard code `1024`. `1024` is only a limit for our GPU in the CI if I remember correctly A100+ support 2048 threads.

Reply:
Done.